lib/Dancer2/Core/Request.pm

```diff
@@ -653,6 +653,14 @@ sub _shallow_clone {
     return $new_request;
 }
 
+
+sub route {
+    my ($self, $route) = @_;
+    if ($route) { $self->{route_matched} = $route; }
+
+    return $self->{route_matched};
```

Comment:
State is the problem. Dancer2 is able to optimize some things based on attribute values never changing. We're using it in several places and I want to use it in more. If the state changes here, we have to account for it.

I was surprised by `forward` so I look. The code seems to create a new request. We do a shallow clone of the existing request and then update the cookies. This might be wrong in general. However, it does combine new options, so the new option could be the next route. This means we won't be breaking `forward`.

Let me try and hack on it when I get home.